phantomgames/GameDetails/GameDetailRepository.swift

```diff
@@ -4,5 +4,17 @@
 //
 //  Created by Aphiwe Shozi on 2024/05/17.
 //
-#warning("Feature is to be developed in another branch")
+
 import Foundation
+
+protocol GameDetailRepositoryType: AnyObject {
+    func fetchGameDetail(id: Int, completion: @escaping (GameDetailResult))
```

Comment:
```suggestion
    func fetchGameDetail(id: Int, completion: @escaping (GameDetailResult))
```
Same here use a more descriptive name